dask/dataframe/io/parquet/core.py

```diff
@@ -95,6 +95,64 @@ def __call__(self, part):
         )
 
 
+ParquetFunctionWrapper = ReadParquetFunctionWrapper
+
+
+class ToParquetFunctionWrapper:
+    """
+    Parquet Function-Wrapper Class
+
+    Writes a DataFrame partition into a distinct
+    parquet file. When called, the function also
+    requires a ``partition_info`` kwarg.
+    """
+
+    def __init__(
+        self,
+        engine,
+        path,
+        fs,
+        partition_on,
+        write_metadata_file,
+        i_offset,
+        name_function,
+        kwargs_pass,
+    ):
+        self.engine = engine
+        self.path = path
+        self.fs = fs
+        self.partition_on = partition_on
+        self.write_metadata_file = write_metadata_file
+        self.i_offset = i_offset
+        self.name_function = name_function
+        self.kwargs_pass = kwargs_pass
+
+    def __call__(self, df, partition_info=None):
+
+        # Get partition index
+        part_i = (partition_info or {}).get("number")
```

Comment:
```suggestion
    def __call__(self, df, *, partition_info):

        # Get partition index
        part_i = partition_info["number"]
```
I don't think `partition_info` is an optional argument. And `(partition_info or {}).get("number")` wouldn't work; the `part_i + self.i_offset` immediately after would fail if `part_i` was None.

Reply:
Ah , good point - I'm going to go with your [BlockwiseDep suggestion](https://github.com/dask/dask/pull/8487/files#r788329012), so this will need to change to `def __call__(self, df, partition_index):` anyway.